registry/remote/credentials/file_store.go

```diff
@@ -95,3 +99,31 @@ func validateCredentialFormat(cred auth.Credential) error {
 	}
 	return nil
 }
+
+// NewAuthConfig creates an authConfig based on cred.
+func NewAuthConfig(cred Credential) configuration.AuthConfig {
```

Comment:
`NewAuthConfig` is public method but its return value `configuration.AuthConfig` is an internal type